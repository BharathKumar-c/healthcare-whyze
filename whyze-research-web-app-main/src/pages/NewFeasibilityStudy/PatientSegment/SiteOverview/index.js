import { Divider } from 'antd';
import React from 'react';
import { Contact, Initiated, Sites } from '../../../../assets';
import { Text } from '../../../../components/basic';
import PropTypes from 'prop-types';
export default function SiteOverview({
  preferredSitesCount,
  sitesInitiatedCount,
  sitesContactedCount,
}) {
  return (
    <div className="site-overview">
      <Text className="site-overview-title" text="Sites overview" />
      <div className="site-overview-container">
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Sites />
            </div>
            <Text className="site-overview-value" text={preferredSitesCount} />
          </div>
          <Text className="site-overview-text" text="Sites" />
        </div>
        <Divider className="site-overview-divider" type="vertical" />
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Contact />
            </div>
            <Text className="site-overview-value" text={sitesContactedCount} />
          </div>
          <Text className="site-overview-text" text="Contacted" />
        </div>
        <Divider className="site-overview-divider" type="vertical" />
        <div className="site-overview-alignment">
          <div className="site-overview-vertical-alignment">
            <div className="site-overview-icon">
              <Initiated />
            </div>
            <Text className="site-overview-value" text={sitesInitiatedCount} />
          </div>
          <Text className="site-overview-text" text="Initiated" />
        </div>
      </div>
    </div>
  );
}

SiteOverview.propTypes = {
  preferredSitesCount: PropTypes.number,
  sitesInitiatedCount: PropTypes.number,
  sitesContactedCount: PropTypes.number,
};
SiteOverview.defaultProps = {
  preferredSitesCount: 0,
  sitesInitiatedCount: 0,
  sitesContactedCount: 0,
};
