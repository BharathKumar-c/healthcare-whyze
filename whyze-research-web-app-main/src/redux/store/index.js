import { configureStore } from '@reduxjs/toolkit';
import reducer from '../reducer/combinedReducer';

const store = configureStore({
  reducer: reducer,

  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export default store;
