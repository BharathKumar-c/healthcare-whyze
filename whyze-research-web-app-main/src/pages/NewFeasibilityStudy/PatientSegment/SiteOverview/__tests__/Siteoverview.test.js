import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../../../../redux/store';
import React from 'react';
import SiteOverview from '..';

function renderWithContext(element) {
  const { container, getByText, getByRole, getByLabelText } = render(
    <Provider store={store}>{element}</Provider>,
  );
  return { container, store, getByText, getByRole, getByLabelText };
}
describe('create tests for SiteOverview', () => {
  test('create snapshot for SiteOverview', () => {
    const { container } = renderWithContext(<SiteOverview />);
    expect(container).toMatchSnapshot();
  });
});
