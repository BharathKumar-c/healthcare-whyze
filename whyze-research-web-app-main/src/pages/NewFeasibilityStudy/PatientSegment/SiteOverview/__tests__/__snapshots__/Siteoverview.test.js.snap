// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create tests for SiteOverview create snapshot for SiteOverview 1`] = `
<div>
  <div
    class="site-overview"
  >
    <span
      class="ant-typography site-overview-title"
    >
      Sites overview
    </span>
    <div
      class="site-overview-container"
    >
      <div
        class="site-overview-alignment"
      >
        <div
          class="site-overview-vertical-alignment"
        >
          <div
            class="site-overview-icon"
          >
            <svg>
              Sites.svg
            </svg>
          </div>
          <span
            class="ant-typography site-overview-value"
          >
            3
          </span>
        </div>
        <span
          class="ant-typography site-overview-text"
        >
          Sites
        </span>
      </div>
      <div
        class="ant-divider ant-divider-vertical site-overview-divider"
        role="separator"
      />
      <div
        class="site-overview-alignment"
      >
        <div
          class="site-overview-vertical-alignment"
        >
          <div
            class="site-overview-icon"
          >
            <svg>
              Contact.svg
            </svg>
          </div>
          <span
            class="ant-typography site-overview-value"
          >
            0
          </span>
        </div>
        <span
          class="ant-typography site-overview-text"
        >
          Contacted
        </span>
      </div>
      <div
        class="ant-divider ant-divider-vertical site-overview-divider"
        role="separator"
      />
      <div
        class="site-overview-alignment"
      >
        <div
          class="site-overview-vertical-alignment"
        >
          <div
            class="site-overview-icon"
          >
            <svg>
              Initiated.svg
            </svg>
          </div>
          <span
            class="ant-typography site-overview-value"
          >
            0
          </span>
        </div>
        <span
          class="ant-typography site-overview-text"
        >
          Initiated
        </span>
      </div>
    </div>
  </div>
</div>
`;
